atoma-proxy/src/server/middleware.rs

```diff
@@ -824,124 +993,158 @@ pub(crate) mod auth {
+        tx_digest: Option<TransactionDigest>,
+        user_id: i64,
+        endpoint: &str,
+    ) -> Result<Request<Body>> {
+        let ProcessedRequest {
+            node_address,
+            signature,
+        } = auth::process_selected_stack(
+            state,
+            &mut req_parts.headers,
+            body_json,
+            selected_node_id,
+            endpoint,
+        )
+        .await?;
+
+        let stack_small_id_header = HeaderValue::from_str(&selected_stack_small_id.to_string())
+            .map_err(|e| AtomaProxyError::InternalError {
+                message: format!("Failed to convert stack small id to header value: {}", e),
+                endpoint: req_parts.uri.path().to_string(),
+            })?;
+        let signature_header =
+            HeaderValue::from_str(&signature).map_err(|e| AtomaProxyError::InternalError {
+                message: format!("Failed to convert signature to header value: {}", e),
+                endpoint: req_parts.uri.path().to_string(),
+            })?;
+        let content_length_header = HeaderValue::from_str(&body_json.to_string().len().to_string())
+            .map_err(|e| AtomaProxyError::InternalError {
+                message: format!("Failed to convert content length to header value: {}", e),
+                endpoint: req_parts.uri.path().to_string(),
+            })?;
+        req_parts
+            .headers
+            .insert(constants::SIGNATURE, signature_header);
+        req_parts
+            .headers
+            .insert(constants::STACK_SMALL_ID, stack_small_id_header);
+        req_parts
+            .headers
+            .insert(CONTENT_LENGTH, content_length_header);
+        if let Some(tx_digest) = tx_digest {
+            let tx_digest_header =
+                HeaderValue::from_str(&tx_digest.base58_encode()).map_err(|e| {
+                    AtomaProxyError::InternalError {
+                        message: format!("Failed to convert tx digest to header value: {}", e),
+                        endpoint: req_parts.uri.path().to_string(),
                     }
                 })?;
-                authenticate_and_process(
-                    request_model,
-                    &state.0,
-                    req_parts.headers.clone(),
-                    body_json,
-                    endpoint == CONFIDENTIAL_IMAGE_GENERATIONS_PATH,
-                    endpoint,
-                )
-                .await
-            }
-            _ => Err(AtomaProxyError::NotFound {
-                message: format!("Invalid or unsupported endpoint: {}", endpoint),
-                endpoint: endpoint.to_string(),
-            }),
+            req_parts
+                .headers
+                .insert(constants::TX_DIGEST, tx_digest_header);
         }
+        let request_model = body_json.get("model").and_then(|m| m.as_str()).ok_or(
+            AtomaProxyError::InvalidBody {
+                message: "Model not found".to_string(),
```

Comment:
you can reuse the above constant over here as well

Reply:
Addressed